From 7e9c82c499592f518e45cf9d049529470971dac7 Mon Sep 17 00:00:00 2001
From: Matthias Reichl <hias@horus.com>
Date: Fri, 2 Feb 2018 20:30:42 +0100
Subject: [PATCH 215/454] ASoC: iqaudio-dac: fix S24_LE format

Remove set_bclk_ratio call so 24-bit data is transmitted in
24 bclk cycles.

Signed-off-by: Matthias Reichl <hias@horus.com>
---
 sound/soc/bcm/iqaudio-dac.c | 18 ------------------
 1 file changed, 18 deletions(-)

--- a/sound/soc/bcm/iqaudio-dac.c
+++ b/sound/soc/bcm/iqaudio-dac.c
@@ -43,18 +43,6 @@ static int snd_rpi_iqaudio_dac_init(stru
 	return 0;
 }
 
-static int snd_rpi_iqaudio_dac_hw_params(struct snd_pcm_substream *substream,
-	struct snd_pcm_hw_params *params)
-{
-	struct snd_soc_pcm_runtime *rtd = substream->private_data;
-	struct snd_soc_dai *cpu_dai = rtd->cpu_dai;
-
-	unsigned int sample_bits =
-		snd_pcm_format_physical_width(params_format(params));
-
-	return snd_soc_dai_set_bclk_ratio(cpu_dai, sample_bits * 2);
-}
-
 static void snd_rpi_iqaudio_gpio_mute(struct snd_soc_card *card)
 {
 	if (mute_gpio) {
@@ -109,11 +97,6 @@ static int snd_rpi_iqaudio_set_bias_leve
 	return 0;
 }
 
-/* machine stream operations */
-static struct snd_soc_ops snd_rpi_iqaudio_dac_ops = {
-	.hw_params = snd_rpi_iqaudio_dac_hw_params,
-};
-
 static struct snd_soc_dai_link snd_rpi_iqaudio_dac_dai[] = {
 {
 	.cpu_dai_name	= "bcm2708-i2s.0",
@@ -122,7 +105,6 @@ static struct snd_soc_dai_link snd_rpi_i
 	.codec_name	= "pcm512x.1-004c",
 	.dai_fmt	= SND_SOC_DAIFMT_I2S | SND_SOC_DAIFMT_NB_NF |
 				SND_SOC_DAIFMT_CBS_CFS,
-	.ops		= &snd_rpi_iqaudio_dac_ops,
 	.init		= snd_rpi_iqaudio_dac_init,
 },
 };
